joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

velocity_controller:
  type: "diff_drive_controller/DiffDriveController"
  left_wheel: "back_left_wheel_link"
  right_wheel: "back_right_wheel_link"
  publish_rate: 50 # default: 50
  pose_covariance_diagonal:
    [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal:
    [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

  # Wheel separation and diameter. These are both optional.
  # diff_drive_controller will attempt to read either one or both from the
  # URDF if not specified as a parameter
  #wheel_separation : 0.28
  #wheel_radius : 0.04688

  enable_odom_tf: true
  publish_cmd: true

  # Wheel separation and radius multipliers
  wheel_separation_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier: 1.0 # default: 1.0

  # Velocity commands timeout [s], default 0.5
  cmd_vel_timeout: 0.25

  # Base frame_id
  base_frame_id: chassis_link #default: base_link

  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits: true
      max_velocity: 1.0 # m/s
      min_velocity: -0.5 # m/s
      has_acceleration_limits: true
      max_acceleration: 0.2 # m/s^2
      min_acceleration: -0.2 # m/s^2
      has_jerk_limits: true
      max_jerk: 5.0 # m/s^3
  angular:
    z:
      has_velocity_limits: true
      max_velocity: 3.0 # rad/s
      has_acceleration_limits: true
      max_acceleration: 0.5 # rad/s^2
      has_jerk_limits: true
      max_jerk: 2.5 # rad/s^3
